### DISCUSSION TITLE
how to handle my own `id` column in Watermelon schema & model?

### DISCUSSION THREAD

**Discussion by lucksp** [Nozbe/WatermelonDB#1849] [Q&A] [answered]
[According to the docs](https://watermelondb.dev/docs/Schema#special-columns): 
> you cannot declare a column named `id` yourself. 

However, my Supabase data tables have `id` as column identifiers, which I need to use.  How do I consume/identify my `id` columns v the Watermelon `id`?

**Comment by Yusuf-Munir** [comment, marked as answer]
Personally I would add a `server_id` column to the table and populate the Supabase id in there.

**Reply by lucksp** [reply]
OK, so there's both `id` and `server_id` in my database.  My schema setup would use `server_id` instead of original `id`?  And then when I query against it in my UI component, I would use `server_id` instead?

**Reply by Yusuf-Munir** [reply]
Yep that's correct.

If you want to query items on the frontend using the Supabase id (which we called `server_id` in your WatermelonDb schema) then you can fetch items like this `tableCollections.query(Q.where('server_id', Q.eq(stringhere));`

So on WatermelonDb schema you will still have the default `id` which WatermelonDB creates and populates, and you will also have your own `server_id` which you will get from Supabase syncing.

And on Supabase schema you can change the name of your primary key to `server_id` to make it easier to sync so you don't get confused. 
Also, you don't have to push or pull your Watermelon DB default ID to and from Supabase.

Just make sure your pull and push functions both use at the `server_id` variable instead of `id`
